pkgs/development/libraries/optparse-bash/default.nix

```diff
@@ -0,0 +1,39 @@
+{ stdenvNoCC
+, lib
+, fetchFromGitHub
+, gnused
+}:
+let
+  pname = "optparse";
+  version = "unstable-20170611";
+in stdenvNoCC.mkDerivation {
+  inherit pname version;
+
+  src = fetchFromGitHub {
+    owner = "nk412";
+    repo = pname;
+    rev = "0dfc5a3344feafc818b1fb051e4bebd195d18893";
+    sha256 = "sha256-fFDKU4ZQxrDokxwjHpXQaL3XGwdrSPugQSJbXMi6tJY=";
+  };
+
+  propagatedBuildInputs = [ gnused ];
```

Comment:
There is just one sed in the file https://github.com/nk412/optparse/blob/master/optparse.bash#L165 which we can replace.

Reply:
What is the recommended substitude?

Reply:
like this https://github.com/NixOS/nixpkgs/blob/cbfd088d28e2b4b1d44a126823ae55c6a463e3dd/pkgs/applications/audio/a2jmidid/default.nix#L21

Reply:
> Would substituteInPlace be available at run time?

Nope, only at build time.

Reply:
Just found that it not only needs `sed`, but `printf` and `awk`, and they all need to be patched manually to optparse.bash by `substituteInPlace`ing with executable paths.
These deps are not documented in the project README and is not noticeable when running from nix-shell.